test/IRGen/opt_in_reflection_metadata.swift

```diff
@@ -0,0 +1,102 @@
+// RUN: %target-swift-frontend -O -enable-opt-in-reflection-metadata -emit-ir %s | %FileCheck %s --check-prefix=CHECK-REL
+// RUN: %target-swift-frontend -enable-opt-in-reflection-metadata -emit-ir %s | %FileCheck %s --check-prefix=CHECK-DEB
+
+// reflection metadata field descriptor opt_in_reflection_metadata.RefProtocol
+// CHECK-REL-DAG: @"$s26opt_in_reflection_metadata11RefProtocol_pMF" = internal constant {{.*}} section "{{[^"]*swift5_fieldmd|.sw5flmd\$B}}
+// CHECK-DEB-DAG: @"$s26opt_in_reflection_metadata11RefProtocol_pMF" = internal constant {{.*}} section "{{[^"]*swift5_fieldmd|.sw5flmd\$B}}
```

Comment:
Protocols should never emit field descriptors because they aren't types by themselves and never have fields. It looks like an oversight that these get emitted in the first place and hopefully nothing will break if we stop emitting them because they're always empty.

Reply:
@Azoy points out that it isn't even possible to find these field descriptors from the protocol descriptor, so yeah, they're probably unused.

Reply:
Not sure, but is it possible that those symbols are used by the debugger? I tried to stop emitting field descriptor for protocols, but several tests started failing (like this one - [`reflect(object: TestGeneric(D() as P))`](https://github.com/apple/swift/blob/main/validation-test/Reflection/reflect_existential.swift#L58)). Not sure how to check what exactly uses it.

Reply:
Protocol "field descriptors" encode whether the protocol is ObjC or not.